### ISSUE TITLE
since 7.1.1 a query will not find documents when an index exists but is not used

### ISSUE THREAD

**Issue by pubkey** [pouchdb/pouchdb#7810]
### Issue
Since version 7.1.1 a query will return no documents, when there is an index used.

### Info
- Environment: Node.js
- Platform: Linux
- Adapter: memory

### Reproduce

I created this test which fails ~9 out of 10 times. I think it is a timing problem~. With the older pouchdb-version, it works.

```js
const PouchDB = require('pouchdb-core');
PouchDB.plugin(require('pouchdb-find'));
PouchDB.plugin(require('pouchdb-adapter-memory'));
const db = new PouchDB(
    'things',
    {
        adapter: 'memory'
    }
);
await db.createIndex({
    index: {
        fields: [
            'passportId'
        ]
    }
});
await db.put({
    _id: 'foobar',
    passportId: 'z3i7q29g4yr1',
    firstName: 'Edison',
    lastName: 'Keebler',
    age: 24
});

const docs = await db.find({
    selector: {
        _id: {}
    },
    limit: 1
});


console.dir(docs); // <- empty array
assert.equal(docs.docs.length, 1);
```

**Comment by karlwestin** [comment]
there seems to have been some change in how things work, but there also seems you're not actually using the index you create in the example above

maybe that's happened between the versions?
see console output from 7.0.0 here: (**note the warning about no index**)
![Screen Shot 2019-06-20 at 9 36 05 AM](https://user-images.githubusercontent.com/359947/59830001-ffc70800-933e-11e9-9c94-7a95fe0bfa98.png)

and the console output from 7.1.1 here:
![Screen Shot 2019-06-20 at 9 36 11 AM](https://user-images.githubusercontent.com/359947/59830024-0e152400-933f-11e9-931b-26c81e7d6d94.png)

Here are links to the JSbins
http://jsbin.com/juhinawire/1/edit?html,js,output
http://jsbin.com/pewitofuma/1/edit?html,js,output

**Comment by pubkey** [comment]
@karlwestin thanks for investigation.
Yes, my query does not use the index. But I think having an index should not affect the result of an unrelated query.

**Comment by karlwestin** [comment]
ok that sounds fair but how about this –  
if you use the index that you create, do you still have the bug or does it only show when not using the index?

**Comment by garrensmith** [comment]
This is kind of weird. The only change we have to pouchdb-find is https://github.com/pouchdb/pouchdb/commit/8aed67088b32f972149d52ba74ca5db044fc3174

Do you get the same issue when running the test in the browser. Also does it pass with 7.0.0 if you use a newer version of leveldown and those adapters. I kind of think this might be an error or change of behaviour in leveldown rather than with pouchdb-find.

**Comment by pubkey** [comment]
@karlwestin If I use the index in the query, I get the correct results [jsbin](https://jsbin.com/dulifuhexu/5/edit?html,js,output). This would be a workaround but I still think that it is wrong behavior of pouchdb. 

@garrensmith Yes the same problem occurs with the memory-adapter in nodejs and also with the idb-adapter in the browser.
When I downgrade pouchdb-find to 7.0.0, I get the correct result [jsbin](https://jsbin.com/niharogiki/1/edit?html,js,output)

**Comment by garrensmith** [comment]
Thanks Pubkey, very strange I will have to look into it


________________________________

**Comment by pubkey** [comment]
@garrensmith Thanks, I will investigate more into version-changes and try to help.

**Comment by pubkey** [comment]
More things I found: 
- Using the older version of the memory-adapter does **not** fix this
- Downgrading to leveldown@4.0.2 does **not** fix this
- Downgrading to leveldown@5.0.1 does **not** fix this

**Comment by pubkey** [comment]
@garrensmith any update on this?
This makes pouchdb unusable for me because it is returning 'false' answers to queries, based on a side-effect of the existing index.

**Comment by garrensmith** [comment]
Sorry. I won’t have time to work on this for awhile.


________________________________

**Comment by garrensmith** [comment]
An you create a reproduce-able test case


________________________________

**Comment by pubkey** [comment]
> An you create a reproduce-able test case
Is my test ok or should I improve it?

**Comment by pubkey** [comment]
Is my test ok or should I improve it?
I think https://github.com/pouchdb/pouchdb/commit/8aed67088b32f972149d52ba74ca5db044fc3174 is the problem. @daleharvey do have any hint if your commit can have caused this?

**Comment by garrensmith** [comment]
@pubkey the best way to help with this is to start a branch with a failing test that proves the issue. We can then explore a few options to try and fix it.

**Comment by pubkey** [comment]
@garrensmith I could reproduce this now on #7895. Please have a look

**Comment by pubkey** [comment]
Still valid

**Comment by pubkey** [comment]
Still valid.

Please remove this useless bot.

**Comment by pubkey** [comment]
No

**Comment by qinyang912** [comment]
No

**Comment by mardii** [comment]
@pubkey Can I ask you something?

In the example above you have your selector defined as:
```javascript
selector: {
  _id: {}
}
```
I don't get the syntax of the query. If the selector for `_id` is an empty object, wouldn't you expect to have 0 results back? It wouldn't match to any document would it? 
At least that is my naïve understanding of the syntax.

I tried both the jsbin you linked above, and when I do:
```javascript
selector: {
  _id: 'foobar'
}
```

I get the document returned (using the default `_id` index I presume). Also the following (querying on a field that isn't indexed):
```javascript
selector: {
  age: 24
}
```
Gives me back the correct document.

I'm asking because I was looking at this issue after reading your release notes on RxDB (cool stuff!)
Lemme know how I can help.

**Comment by pubkey** [comment]
@mardii thanks for the investigation.
It seems that I was too deep in the tunnel when I created my test.
I have improved the test at #8066 to better show what I think is a bug. Is it understandable now?

When two databases are used where both have the same data but one has an index. Then the query returns different results. I think this is a bug because having and index is a side effect, especially when the index is not relevant for the query (here `passportId`). So having that index is not something that should affect which documents must be returned for a given query.
Before 8aed67088b32f972149d52ba74ca5db044fc3174 this worked and both databases returned the same results.

My example query might be strange but the whole pouchdb-find does not work anymore with the test suite of RxDB because it returns wrong results in many different parts. I picked this query because it was the first one where I could "prove" the wrongness.

**Comment by mardii** [comment]
So I've been banging my head against this the whole day.
@pubkey Cheers for changing the tests and explaining it.

I'm still in the middle of testing a lot, trying to figure out where the problem lies.
It doesn't seem to be an issue with the code differenced of the mentioned change 8aed670. I get the same inconsistencies when undoing those specific changes.
I have come through most of the query-planner, but the issue doesn't seem to be there.

Tomorrow I will try to open up yet another PR to show my tests (I've added more scenario's and moved the test to the find test suites).

I might add inconstancies only popup in the edge cases
```
selector: {
}
```
and
```
selector: {
  _id: {}
}
```

To be continued.

**Comment by daleharvey** [comment]
Fixed in ac03f8a
2a06404

**Comment by stevenroussey-privicy** [comment]
Will there be a new release soon with this fix?
